R/drive_mv.R

```diff
@@ -1,60 +1,119 @@
 #' Move Google Drive file.
 #'
+#' Move a Drive file to a different folder, give it a different name, or both.
+#'
 #' @template file
 #' @template path
+#' @param name Character, new file name if not specified as part of `path`. Any
+#'   name obtained from `path` overrides this argument. Defaults to current
+#'   name.
 #' @template verbose
 #'
 #' @template dribble-return
 #' @export
-drive_mv <- function(file = NULL,
-                     path = NULL,
-                     verbose = TRUE) {
+#' @examples
+#' \dontrun{
+#' ## create a file to move
+#' file <- drive_upload(system.file("DESCRIPTION"), "DESC")
+#'
+#' ## rename it, but leave in current folder (root folder, in this case)
+#' file <- drive_mv(file, "DESC-renamed")
+#'
+#' ## create a folder to move the file into
+#' folder <- drive_mkdir("new-folder")
+#'
+#' ## move the file and rename it again,
+#' ## specify destination as a dribble
+#' file <- drive_mv(file, path = folder, name = "DESC-re-renamed")
+#'
+#' ## verify renamed file is now in the folder
+#' drive_ls(folder)
+#'
+#' ## move the file back to root folder
+#' file <- drive_mv(file, "~/")
+#'
+#' ## move it again
+#' ## specify destination as path with trailing slash
+#' ## to ensure we get a move vs. renaming it to "new-folder"
+#' file <- drive_mv(file, "new-folder/")
+#' }
+drive_mv <- function(file = NULL, path = NULL, name = NULL, verbose = TRUE) {
   file <- as_dribble(file)
   file <- confirm_single_file(file)
-  folder <- as_dribble(path)
-  folder <- confirm_single_file(folder)
-
   if (!is_mine(file)) {
```

Comment:
You know, there are probably many other places where we have comparable need to check "is it mine?" 🤔